jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -609,6 +610,48 @@ ecma_builtin_typedarray_prototype_filter (ecma_value_t this_arg, /**< this argum
 } /* ecma_builtin_typedarray_prototype_filter */
 
 /**
+ * The %TypedArray%.prototype object's 'reverse' routine
+ *
+ * See also:
+ *          ES2015, 22.2.3.21
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_typedarray_prototype_reverse (ecma_value_t this_arg) /**< this argument */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_arg);
+  uint32_t len = ecma_typedarray_get_length (obj_p);
+  ecma_object_t *arraybuffer_p = ecma_typedarray_get_arraybuffer (obj_p);
+  lit_utf8_byte_t *buffer = (ecma_arraybuffer_get_buffer (arraybuffer_p)
+                             + ecma_typedarray_get_offset (obj_p));
+  uint8_t shift = ecma_typedarray_get_element_size_shift (obj_p);
+  uint8_t element_size = (uint8_t) (1 << shift);
+  uint32_t middle = (len / 2) << shift;
+  uint32_t buffer_length = len << shift;
```

Comment:
we could rename buffer_length to buffer_last and:

    uint32_t buffer_last = (len << shift) - element_size

    uint32_t upper = buffer_last - lower